packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts

```diff
@@ -0,0 +1,80 @@
+import { GridSortingModelApplier } from './gridSortingState';
+import {
+  GridApiRef,
+  GridComparatorFn,
+  GridRowId,
+  GridRowTreeNodeConfig,
+  GridSortCellParams,
+  GridSortModel,
+} from '../../../models';
+import { isDesc } from '../../../utils/sortingUtils';
+
+type GridSortingFieldComparator = {
+  getSortCellParams: (id: GridRowId) => GridSortCellParams;
+  comparator: GridComparatorFn;
+};
+
+/**
+ * Generates a method to easily sort a list of rows according to the current sort model.
+ * @param {GridSortModel} sortModel The model with which we want to sort the rows.
+ * @param {GridApiRef} apiRef The API of the grid.
+ * @returns {GridSortingModelApplier | null} A method that sorts a list of rows according to the current sort model. If `null`, we consider that the rows should remain in the order there were provided.
```

Comment:
I've been confused by the returned function. It does not seem to sort the list. It provides the list of row ids sorted according to the sort model